client/src/components/account/AuthContainer.js

```diff
@@ -12,11 +13,53 @@ class AuthContainer extends Component {
       username: '',
       password: '',
       passwordCheck: '',
+      newPass: '',
+      newPassCheck: '',
       errorMessage: ''
     }
 
     this.attemptLogin = this.attemptLogin.bind(this);
     this.attemptRegister = this.attemptRegister.bind(this);
+    this.attemptPassChange = this.attemptPassChange.bind(this);
+  }
+
+  attemptPassChange(e) {
+    e.preventDefault()
+    if (this.state.username && this.state.password && this.state.newPass && this.state.newPassCheck) {
+      if (this.state.newPass === this.state.newPassCheck) {
+        fetch('/changepass/', {
+          method: 'PUT',
+          body: JSON.stringify({ username: this.state.username, password: this.state.password, newPass: this.state.newPass}),
+          headers: new Headers({ 'Content-type': 'application/json' })
+        })
+        .then((response) => {
+          if (response.ok) {
+            return response.json();
+          } else {
+            //TODO handle error messages here
+            console.log(response);
+          }
+        })
+        .then((data) => {
+          if (data.token) {
+            localStorage.setItem('jwt', data.token);
```

Comment:
let's have the server respond with status 200 if it succeeds, and then have the app transition to a "You successfully changed your password" page like successful registration.
```
.then(response=> { 
  if (response.ok) {
    this.props.switchMode('passwordChangeSuccess')
  } else {
    this.setState({errorMessage: ...})
}
}
```